src/payment/strategies/klarna/klarna-payment-strategy.ts

```diff
@@ -29,7 +31,13 @@ export default class KlarnaPaymentStrategy implements PaymentStrategy {
     ) {}
 
     initialize(options: PaymentInitializeOptions): Promise<InternalCheckoutSelectors> {
-        return this._klarnaScriptLoader.load()
+        const { klarnav2 } = options;
+
+        if (klarnav2) {
+            return this._initializeV2(options);
```

Comment:
You should create a new strategy for V2 instead of adding a bunch of `if` conditions to make V1 work with V2 as well. There will be some duplications but it's more important to keep the separation than to keep the code DRY. If you want to re-use part of the code, you can extract some common functionalities into separate classes or functions and compose them with the V1 and V2 strategies.

The API should also return `gateway` as `klarnav2` to avoid name collision with V1.